### PR TITLE
Adding props for Dialog ariaLabelledById and ariaDescribedById

### PR DESCRIPTION
**Pull request by zoarif**
#### Description of changes

This adds two props for Dialog ariaLabelledById and ariaDescribedById. The idea to add these is to workaround an issue where JAWS does not speak out ariaDescribedBy so we need to append the complete string in aria-Labelledby. 

#### Focus areas to test

I have added examples that tests these props.

### CODE REVIEW

**Comment by benkaiser** [comment]
👍

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import * as React from 'react';
 import {
   BaseComponent,
   css,
-  getId
+  getId,
+  autobind
```
Comment (antonlabunets) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx]: please sort by alphabet

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -83,8 +87,8 @@ export class Dialog extends BaseComponent<IDialogProps, IDialogState> {
         onDismissed={ onDismissed }
         onLayerDidMount={ onLayerDidMount }
         responsiveMode={ responsiveMode }
-        subtitleAriaId={ subText && id + '-subText' }
-        titleAriaId={ title && id + '-title' }
+        subtitleAriaId={ this._getAriaLabelId(ariaDescribedById, subText, id + 'subText') }
```
Comment (antonlabunets) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx]: I think you have to bring back - before `subText` to keep backward compatible, don't you?

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -101,4 +105,13 @@ export class Dialog extends BaseComponent<IDialogProps, IDialogState> {
       </Modal>
     );
   }
+
+  @autobind
+  private _getAriaLabelId(ariaId: string, text: string, alternativeId: string): string {
+    if (ariaId) {
```
Comment (antonlabunets) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx]: `return ariaId || text && alternativeId`?

**Comment by antonlabunets** [comment]
See if you can add to unit tests too.

**Comment by micahgodbolt** [comment]
Hey @zoarif and @antonlabunets, just a heads up that this PR got pulled in a bit prematurely, and that we were planning on resolving this by separating content and modal props in #1865

I'll make sure that these 2 props are covered, but just be aware that I'll be deprecating these 2 props, and you'll be able to pass them in via contentProps in the near future. 

cc @dzearing

**Pull request merged**